pkg/stub/handler.go

```diff
@@ -69,16 +69,56 @@ func (h *Handler) EnsureDefaultClusterIngress() error {
 		return err
 	}
 
-	err = sdk.Create(ci)
-	if err == nil || errors.IsAlreadyExists(err) {
-		if err == nil {
-			logrus.Infof("created default cluster ingress %s/%s", ci.Namespace, ci.Name)
+	changed, nci, err := checkClusterIngress(ci)
+	if err != nil {
```

Comment:
I think this can use: clusterIngressStatusEqual
from PR: #38

Reply:
Maybe not `clusterIngressStatusEqual`, but the part that compares `.loadBalancer.Ingress` could be factored into a function that could be used both by `checkClusterIngress` and by `clusterIngressStatusEqual`.  Now that #47 is merged, I'll rebase #38 Tuesday and factor out a `LoadBalancerStatusEqual` function.  No need for that to block this PR though; I can change `checkClusterIngress` to use `LoadBalancerStatusEqual` if #48 merges before #38 is ready.

Reply:
Err, actually, `checkClusterIngress` is comparing only `.spec.ingressDomain` (a string) whereas `clusterIngressStatusEqual` is comparing `.status.loadBalancer` (a slice of a struct that has two string values), so there is really no opportunity for code re-use here as far as I see.

Reply:
I might be confused but if the LoadBalancerIngress struct are different as being checked in your PR [1]  it means there's an update it should be enough for this purpouse.

[1] https://github.com/openshift/cluster-ingress-operator/pull/38/files#diff-9ae69f8dcadd8df44e0b10265f3a91a8R107